### PR TITLE
fix(api): Create workaround for null union with multiple types

### PR DESCRIPTION
**Pull request by schew2381**
See comment in code for detail

### CODE REVIEW

**Review by nhsiehgit — APPROVED** [review]
🙌 nice! Thanks for figuring this out!

File: src/sentry/apidocs/spectacular_ports.py [hunk 1]
```diff
@@ -146,7 +146,16 @@ def resolve_type_hint(hint) -> Any:
         else:
             schema = resolve_type_hint(type_args[0])
         if type(None) in args:
-            schema["nullable"] = True
+            # There's an issue where if 3 types are OR'd together and one of
```
Comment (JoshFerge) [inline src/sentry/apidocs/spectacular_ports.py]: wanna double check, what does this function look like on drf-spectacular's main branch now? do we need this shim anymore now that we're on an up to date python version?

**Review by JoshFerge — CHANGES_REQUESTED** [review]
requesting changes until we verify the drf-spectacular's upstream doessn't already have this fix

File: src/sentry/apidocs/spectacular_ports.py (hunk 1 above)
Comment (schew2381) [reply src/sentry/apidocs/spectacular_ports.py]: Hmm I think we might be able to use the original now. The [typeddict part](https://github.com/tfranzel/drf-spectacular/blob/master/drf_spectacular/plumbing.py#L1274-L1287) looks like it'll work, the only difference is it doesn't look like it excludes fields like ours. in `_resolve_typeddict`?

It seems the very recent version also now supports openapi 3.1 which drops `nullable` in favor of `null`, although I don't see the code doing this in the Union check in `resolve_type_hint` 🤔

**Review by JoshFerge — APPROVED** [review]

**Review by JoshFerge — COMMENTED** [review]
thanks @schew2381 for checking. if you get time, could be worth trying to contribute the "excluded_fields" change upstream, and perhaps showing what we're doing to attempt to fix in that thread.

if we can get both of those things figured out, would be lovely to be able to get rid of this shim!

**Pull request merged**